aws_utils/emr/statuschecker.py

```diff
@@ -0,0 +1,93 @@
+import time
+import boto3
+import logging
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class NoSuchActivityError(Exception):
+    """
+    A custom exception
+    """
+
+
+def get_pipeline_state(pipeline_name, region):
+    """
+    Connects to AWS and gets a status of a pipeline
+    :param pipeline_name: (str) name of the Pipeline
+    :param region: (str) AWS region
+    :return: current status of a Pipeline
+    Possible return values : ACTIVATING, CANCELED, CASCADE_FAILED, DEACTIVATING, FAILED, FINISHED, INACTIVE, PAUSED,
+                             PENDING, RUNNING, SHUTTING_DOWN, SKIPPED, TIMEDOUT, VALIDATING, WAITING_FOR_RUNNER, WAITING_ON_DEPENDENCIES
+    """
+    # create a client connection
+    boto3client_pipelines = boto3.client('datapipeline', region) #'us-east-1')
+    pipelines = boto3client_pipelines.list_pipelines()
+
+    current_status_pipeline = ''
+    # get the status of the pipeline
+    for pipeline in pipelines['pipelineIdList']:
```

Comment:
It might be useful to check if the number of pipelines matched is 1, and raise exceptions if it is zero or >1.